### ISSUE TITLE
Inconsistency between micro_train_batch_size and train_batch_size

### ISSUE THREAD

**Issue by zkshan2002** [OpenRLHF/OpenRLHF#459]
Hi, thanks for your great work. I am curious about micro_train_batch_size and train_batch_size args in PPO experiment.
In my knowledge, micro_train_batch_size is the amount of data consumed on each gpu for a gradient step, thus micro_train_batch_size * num_gpu yield the actual global training batch size, there is not need to specify train_batch_size. In the PPO code, train_batch_size is only used to initialize a scheduler. But when I pass in an inconsistent train_batch_size(not equal to micro_train_batch_size * num_gpu), the learning rate curve in my PPO exp still looks fine.
I wonder why is the param train_batch_size free to specify, instead of set fixed to micro_train_batch_size * num_gpu? Is there any trouble the inconsistency may rise that I didn't notice?
Thanks in advance for your reply.

**Comment by zkshan2002** [comment]
I learned about gradient accumulation. It was handled implicitly in Deepspeed so I didn't notice previously. I'll close this issue.
